playwright/src/test/java/com/microsoft/playwright/TestRouteWebSocket.java

```diff
@@ -0,0 +1,180 @@
+  @AfterAll
+  void stopWebSockerServer() throws IOException, InterruptedException {
+    webSocketServer.stop();
+  }
+
+  @AfterEach
+  void resetWebSocketServer() {
+    webSocketServer.reset();
+  }
+
+  private void setupWS(Page target, int port, String binaryType) {
+    setupWS(target.mainFrame(), port, binaryType);
+  }
+  private void setupWS(Frame target, int port, String binaryType) {
+    target.navigate("about:blank");
+    target.evaluate("({ port, binaryType }) => {\n" +
+      "    window.log = [];\n" +
+      "    window.ws = new WebSocket('ws://localhost:' + port + '/ws');\n" +
+      "    window.ws.binaryType = binaryType;\n" +
+      "    window.ws.addEventListener('open', () => window.log.push('open'));\n" +
+      "    window.ws.addEventListener('close', event => window.log.push(`close code=${event.code} reason=${event.reason} wasClean=${event.wasClean}`));\n" +
+      "    window.ws.addEventListener('error', event => window.log.push(`error`));\n" +
+      "    window.ws.addEventListener('message', async event => {\n" +
+      "      let data;\n" +
+      "      if (typeof event.data === 'string')\n" +
+      "        data = event.data;\n" +
+      "      else if (event.data instanceof Blob)\n" +
+      "        data = 'blob:' + await event.data.text();\n" +
+      "      else\n" +
+      "        data = 'arraybuffer:' + await (new Blob([event.data])).text();\n" +
+      "      window.log.push(`message: data=${data} origin=${event.origin} lastEventId=${event.lastEventId}`);\n" +
+      "    });\n" +
+      "    window.wsOpened = new Promise(f => window.ws.addEventListener('open', () => f()));\n" +
+      "  }", mapOf("port", port, "binaryType", binaryType));
+  }
+
+  private void setupRoute(Page page, String mock) {
+    if ("no-match".equals(mock)) {
+      page.routeWebSocket(Pattern.compile("/zzz/"), ws -> {});
+    } else if ("pass-through".equals(mock)) {
+      page.routeWebSocket(Pattern.compile("/.*/"), ws -> {
+        WebSocketRoute server = ws.connectToServer();
+        ws.onMessage(message -> {
+          if (message.text() != null) {
+            server.send(message.text());
+          } else {
+            server.send(message.binary());
+          }
+        });
+        server.onMessage(message -> {
+          if (message.text() != null) {
+            ws.send(message.text());
+          } else {
+            ws.send(message.binary());
+          }
+        });
+      });
+    }
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = {"no-mock", "no-match", "pass-through"})
+  public void shouldWorkWithTextMessage(String mock, Page page) throws Exception {
+    setupRoute(page, mock);
+    Future<WebSocket> wsPromise = webSocketServer.waitForWebSocket();
+    setupWS(page, webSocketServer.getPort(), "blob");
+
+    page.waitForCondition(() -> {
+      Boolean result = (Boolean) page.evaluate("() => window.log.length >= 1");
+      return result;
+    }, new Page.WaitForConditionOptions().setTimeout(10_000));
```

Comment:
Is there a way to set default timeout for `waitForCondition`? If not, we should consider adding that.

Reply:
It respects default timeout, I was setting it to lower value for local testing, removed.